wire.md

```diff
@@ -0,0 +1,1008 @@
+(i.e. `(1 - 1 / (2**32)) ** 43_000_000 > 0.99`)
+
+When generating a new `req_id`, a client MUST discard and generate a new one if
+the one generated collides with a known active `req_id`.
+
+When a client forwards a request to further peers, the `req_id` MUST NOT be
+changed, so that routing loops can be more easily detected by peers in the
+network.
+
+The protocol MAY be extended by implementers by creating additional
+`msg_type`s. Implementers MUST only use `msg_type > 255`. The
+first 256 are reserved for core protocol use.
+
+#### 6.3.2 Requests
+
+##### 6.3.2.1 Header
+
+Every request MUST have its header bytes be proceeded by the following request
+header fields:
+
+field      | type       | desc
+-----------|------------|-----------------------------------
+`ttl`      | `varint`   | number of network hops remaining
+
+The value of `ttl` MUST be between 0 and 16. 0 means "do not forward this
+request to any other peers".
+
+More fields follow below for the different request types, whose bytes MUST
+immediately follow the above message and request header bytes.
+
+##### 6.3.2.2 Post Request
+
+Request a set of posts, given their hashes.
+
+field        | type                | desc
+-------------|---------------------|-------------------------------------
+`hash_count` | `varint`            | number of hashes to request
+`hashes`     | `u8[32*hash_count]` | hashes, concatenated together
+
+`msg_type` MUST be set to `2`.
+
+Results are provided by one or more Post Response messages.
+
+The responder SHOULD immediately return what data is locally available, rather
+than holding on to the request in anticipation of perhaps seeing the requested
+hashes in the future.
+
+##### 6.3.2.3 Cancel Request
+
+Conclude a given `req_id` and stop receiving responses for that request.
+
+This request can be used to terminated previously sent, long-lived requests.
+
+field        | type                | desc
+-------------|---------------------|-------------------------------------
+`cancel_id`  | `u8[4]`             | the `req_id` of the request to be cancelled
+
+Receiving this request indicates that any further responses sent back with a
+`req_id` matching the given `cancel_id` will be ignored and discarded.
+
+`msg_type` MUST be set to `3`.
+
+`cancel_id` MUST be set to the `req_id` of the request to be cancelled.
+
+There MUST NOT be any responses to this request.
+
+Like any other request, this request MUST have its own unique `req_id` in order
+to function as intended. `cancel_id` is used to set the request identifier to
+cancel, not the `req_id`.
+
+A peer receiving a Cancel Request MUST forward it along the same route and
+peers it forwarded the original message with `req_id = cancel_id`, to the same
+peers as the original request, so that all peers who know of the original
+request are notified. This request's `ttl` MUST be ignored, in order to ensure
+all original recipients of the original request are reached.
```

Comment:
Huh, interesting choice to add a new code path for ignoring the ttl rather than just trusting the canceller to not mess it up.

Reply:
Agreed. Removing this.